### ISSUE TITLE
Create a State system

### ISSUE THREAD

**Issue by porowns** [KryptedGaming/krypted#250]
Instead of doing a ton of weird things with groups, I'd like to have a state system similar to that of AllianceAuth. 

Back in the early days, we had the **Guild** system, which was a more primitive state system that allowed guild managers to add people to their guilds, and then players could join and request groups within a guild. This was replaced by group dependency lists later on because the guild system seemed forced, but ultimately it was the best solution and we need to move towards it again. 

Basically, we need **State** objects that act as buckets of groups. A Guest may be able to see guest groups, while an EVE Online state might be able to see EVE Online groups. This will allow us to put people in states based on certain conditions- for example EVE Online corporations or applications. 

From there, I think we should be able to extend state objects like EveStateObject, which will create a state object but also let us extend them with the corporations/etc.

**Comment by porowns** [comment]
@Czahrien do you think you could do a proof of concept of a State model with an EveStateModel extending it? This way we could see if it's the right way forward... Ideally I'd like to keep the state separate from entities like EVE corporations

**Comment by porowns** [comment]
I think a more simple implementation would be the following:

States have: 
* A qualifying group (a group that qualifies a user to be added to this state)
* Enabling groups (what group are accessible by this state) 
* Priority (what the priority of this state is) 

Then there would be two tasks: 
* Task that goes through all users and removes groups that are not enabled by their state
* Task that goes through all users and updates their state based on qualified groups + priority

We would need a "Groups" view after this that allows users to: 
* See a list of groups that they can join or request, with actions for them 
* Allow users with change permission (either in the back-end or front-end) to approve these requests for groups they are apart of 
* Allow users with superuser permission to approve requests for all groups

**Comment by porowns** [comment]
Group requests might just be better suited for a separate package, but they could be here as well.. not sure.

**Comment by porowns** [comment]
https://github.com/KryptedGaming/django-eveonline-group-states
